alert-common/src/main/java/com/synopsys/integration/alert/common/workflow/processor/SummaryMessageContentProcessor.java

```diff
@@ -105,60 +109,78 @@ private ProviderMessageContent summarize(final ProviderMessageContent message) {
+                                                                .map(ComponentItem::getComponentKeys)
+                                                                .map(ComponentKeys::getShallowKey)
+                                                                .collect(Collectors.groupingBy(Function.identity(), Collectors.counting()));
 
-    private String createSummaryKey(ComponentKeys componentKeys, ComponentAttributeMap itemsOfSameName) {
-        StringBuilder summaryKeyBuilder = new StringBuilder();
-        summaryKeyBuilder.append(componentKeys.getShallowKey());
-        for (List<LinkableItem> namedItems : itemsOfSameName.values()) {
-            if (namedItems.size() == 1) {
-                final LinkableItem item = namedItems
-                                              .stream()
-                                              .findAny()
-                                              .orElseThrow();
-                if (!item.isCollapsible() && item.isPartOfKey()) {
-                    summaryKeyBuilder.append(item.getName());
-                    summaryKeyBuilder.append(item.getValue());
+                        LinkableItem countItem = new LinkableItem(COMPONENT_ITEM_NAME_COMPONENTS, String.valueOf(componentCounts.keySet().size()));
+                        countItem.setCollapsible(true);
+                        countItem.setCountable(true);
+                        countItem.setSummarizable(true);
+                        countItem.setPartOfKey(true);
+                        builder.applyComponentAttribute(countItem);
+
+                        final Collection<LinkableItem> summarizedLinkableItems = summarizeComponentItemData(attributeSet);
+                        builder.applyAllComponentAttributes(summarizedLinkableItems);
+                        componentSummaries.add(builder.build());
+                    }
+                } catch (AlertException ex) {
+                    throw new AlertRuntimeException(ex);
                 }
             }
         }
+        return componentSummaries;
+    }
 
-        return summaryKeyBuilder.toString();
+    private Collection<LinkableItem> summarizeComponentItemData(Collection<LinkableItem> attributes) {
+        if (attributes.isEmpty()) {
+            return List.of();
+        }
+        ComponentAttributeMap attributeMap = new ComponentAttributeMap();
+        for (final LinkableItem item : attributes) {
+            final String name = item.getName();
+            attributeMap.computeIfAbsent(name, ignored -> new LinkedList<>()).add(item);
+        }
+        return createSummarizedLinkableItems(attributeMap);
     }
 
-    private SortedSet<LinkableItem> createSummarizedLinkableItems(ComponentAttributeMap componentAttributeMap) {
-        final SortedSet<LinkableItem> summarizedLinkableItems = new TreeSet<>();
+    private Collection<LinkableItem> createSummarizedLinkableItems(ComponentAttributeMap componentAttributeMap) {
```

Comment:
Don't we want the return type to be List here and use Collection for passed args?